common/test/acceptance/pages/lms/courseware.py

```diff
@@ -333,6 +333,46 @@ def is_gating_banner_visible(self):
             and self.q(css='.btn-brand').text[0] == u'Go To Prerequisite Section' \
             and self.q(css='.problem-header').text[0] == u'Content Locked'
 
+    @property
+    def is_word_cloud_rendered(self):
+        """
+        Check for word cloud fields presence
+        """
+        return self.q(css='.input-cloud').present
+
+    def input_word_cloud(self, answer):
+        """
+        Fill the word cloud fields
+        """
+        self.wait_for_element_visibility('.input-cloud', "Word cloud fields are visible")
+        fields = self.q(css='.input-cloud').results
+        total_fields = len(fields)
+        css = '.input_cloud_section  label:nth-child({}) .input-cloud'
+        for index in range(1, total_fields + 1):
+            self.q(css=css.format(index)).fill(answer + str(index))
```

Comment:
```python
def input_word_cloud(self, words):
    """
    Fill the word cloud fields with words

    Arguments:
        words (list),  list of words
    """

    self.wait_for_element_visibility('.input-cloud', "Word cloud fields are visible")
    css = '.input_cloud_section label:nth-child({}) .input-cloud'
    for index in range(1, len(self.q(css='.input-cloud')) + 1):
        self.q(css=css.format(index)).fill(words[index])

    self.save_word_cloud()
```

Reply:
yes, this looks even better. I just had to convert the index variable into the string.